streams/src/test/java/org/apache/kafka/streams/integration/RelaxedNullKeyRequirementJoinTest.java

```diff
@@ -91,13 +101,22 @@ void testRelaxedLeftStreamTableJoin() {
     @Test
     void testRelaxedOuterStreamStreamJoin() {
         leftStream
-            .outerJoin(rightStream, JOINER, WINDOW)
+            .outerJoin(rightStream, JOINER, WINDOW_60MS_GRACE_10MS)
             .to(OUT);
         initTopology();
         right.pipeInput(null, "rightValue", 1);
-        left.pipeInput(null, "leftValue");
+        left.pipeInput(null, "leftValue", 90);
+        left.pipeInput(null, "lateArrival-dropped", 19);
+        right.pipeInput(null, "lateArrival-dropped", 19);
+        right.pipeInput(null, "lateArrivalWithinGrace", 20);
+        left.pipeInput(null, "lateArrivalWithinGrace", 20);
         assertEquals(
-            Arrays.asList(new KeyValue<>(null, "null|rightValue"), new KeyValue<>(null, "leftValue|null")),
+            Arrays.asList(
+                new KeyValue<>(null, "null|rightValue"),
+                new KeyValue<>(null, "leftValue|null"),
+                new KeyValue<>(null, "null|lateArrivalWithinGrace"),
+                new KeyValue<>(null, "lateArrivalWithinGrace|null")
```

Comment:
As above for last two records -- we should not yet seem them, without advancing time to `91`